### PR TITLE
feature: countless pagination

### LINKED ISSUE #1748: Let "total count" be hidden from pagination for large tables.
### Feature

For large tables (with tens of millions of rows), the "total count" query makes the index page extremely slow. [Active Admin provides an option `pagination_total`](https://activeadmin.info/3-index-pages.html) to disable the count query. Is there a similar configuration option in Avo?

```[tasklist]
### Tasks
- [ ] https://github.com/avo-hq/avo/issues/2027
```

### PR DESCRIPTION
**Pull request by Paul-Bob**
# Description

Fixes #1748 

Added new DSL to resource:
```ruby
self.pagination = -> do
  {
    type: :countless, # default -> :normal
    size: []          # default -> [1, 2, 2, 1]
  }
end
```

```ruby
class CourseLinkResource < Avo::BaseResource
  ...
  self.pagination = -> do
    {
      type: :countless,
      size: []
    }
  end

  # OR

  self.pagination = {
    type: :countless,
    size: []
  }
  ...
end
```

image

image

image

image

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
  - https://github.com/avo-hq/avodocs/pull/126 
- [x] I have added tests that prove my fix is effective or that my feature works

## Manual review steps

1. Add `self.pagination = -> do
            {
              type: :countless,
              size: []
            }
          end` to any resource
2. Verify resource's index page

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

**Comment by Paul-Bob** [comment]
We need translations for:

- [ ] avo.ar.yml
- [ ] avo.fr.yml
- [ ] avo.nb.yml
- [ ] avo.nn.yml
- [ ] avo.tr.yml

**Review by adrianthedev — CHANGES_REQUESTED** [review]
Let's change the API from `slef.countles` to 
```ruby
self.pagination = {
  type: :normal/:countless/:countless_minimal
}
```

This will allow us to future-proof the pagination API.

**Comment by Paul-Bob** [comment]
> ```ruby
> countless_minimal
> ```

About the `countless_minimal` we talked that we want to achieve this:

image

The result on the image can be obtained by `countless` and `size: []`, actually

File: spec/system/avo/select_all_spec.rb [hunk 1]
```diff
@@ -10,95 +10,142 @@
   let!(:salmon) { create_list :fish, spec_salmon_number, name: "Spec Salmon" }
   let!(:url) { "/admin/resources/fish?per_page=#{per_page}" }
 
-  describe "without applyed filters" do
-    context "select first page" do
-      it "releases the fish from the selected page" do
-        visit url
+  # describe "without applyed filters" do
```
Comment (adrianthedev) [inline spec/system/avo/select_all_spec.rb:13]: Why did we comment these tests?

**Review by adrianthedev — APPROVED** [review]
Left a few comments. Good job!

File: spec/system/avo/select_all_spec.rb (hunk 1 above)
Comment (Paul-Bob) [reply spec/system/avo/select_all_spec.rb:13]: My bad, I commeted on dev when working on it and forgot to uncomment 🤦🏼

**Comment by adrianthedev** [comment]
Let's use `deafult` instead of `nornmal`

**Pull request merged**